docusaurus/docs/iOS/swiftui/swiftui-cookbook/custom-channel-list.md

```diff
@@ -0,0 +1,237 @@
+---
+title: Custom Channel List
+description: How to build your own UI for previewing channels
+---
+
+There are different variations on how to present a list of channels. Our `ChatChannelListView` component provides a lot of customization options to support those variations.
+
+However, in some cases, you would need to build a completely custom UI. In this cookbook, we will build a simple channel list view, without using our UI components, only the low-level client state.
+
+![Screenshot showing the custom channel list](../../assets/custom-channel-list.png)
+
+## Channels State
+
+We will expose the channels state with a view model, for example `CustomChannelListViewModel`. The view model will be an observable object, and it will implement the `ChatChannelListControllerDelegate`.
+
+We will keep the channels as a published variable in the view model.
+
+```swift
+@Published public var channels = LazyCachedMapCollection<ChatChannel>()
+```
+
+The `ChatChannelListControllerDelegate` has a method called `didChangeChannels`, that sends updates when the channels are updated.
```

Comment:
Is `didChangeChannels` a method or an argument?

Reply:
updated it a bit